hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/repository/FileDataRepository.java

```diff
@@ -26,17 +27,20 @@ public interface FileDataRepository extends CrudRepository<FileData, Long> {
     @Query(
             value =
                     """
-                with latest_create as (
-                      select max(file_data.consensus_timestamp) as consensus_timestamp
-                      from file_data
-                      where file_data.entity_id = ?1 and file_data.transaction_type in (17, 19)
-                    )
-                    select
-                    string_agg(file_data.file_data, '' order by file_data.consensus_timestamp) as file_data
-                    from file_data
-                    join latest_create l on file_data.consensus_timestamp >= l.consensus_timestamp
-                    where file_data.entity_id = ?1 and file_data.transaction_type in (16, 17, 19)
-                      and ?2 >= l.consensus_timestamp""",
+            select * from file_data
+            where file_data.entity_id = ?1
+              and file_data.consensus_timestamp >= (
+                select file_data.consensus_timestamp
+                from file_data
+                where file_data.entity_id = ?1
+                  and file_data.consensus_timestamp <= ?2
+                  and (file_data.transaction_type = 17
+                         or (file_data.transaction_type = 19
+                              and
+                             length(file_data.file_data) <> 0))
+              order by file_data.consensus_timestamp desc
+              limit 1
+            ) and file_data.consensus_timestamp <= ?2""",
```

Comment:
I don't think any of the fully qualified column names are required at all, also missing `order by consensus_timestamp` in the main query

```suggestion
            where entity_id = ?1
              and consensus_timestamp >= (
                select consensus_timestamp
                from file_data
                where entity_id = ?1
                  and consensus_timestamp <= ?2
                  and (transaction_type = 17 or (transaction_type = 19 and length(file_data) <> 0))
              order by consensus_timestamp desc
              limit 1
            ) and consensus_timestamp <= ?2
            order by consensus_timestamp""",
```

Reply:
Thanks I have updated it.